x-pack/test/rule_registry/common/lib/authentication/roles.ts

```diff
@@ -0,0 +1,598 @@
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+/**
+ * ****************************************
+ * These are used for testing update alerts privileges
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ * ****************************************
+ */
+
+export const observabilityMinReadAlertsAll: Role = {
+  name: 'obs_only_alerts_read',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['minimal_read', 'alerts_all'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
+        },
+        spaces: ['space1'],
+      },
+    ],
+  },
+};
+
+export const observabilityMinReadAlertsAllSpacesAll: Role = {
+  name: 'obs_minimal_read_alerts_read_spaces_all',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['minimal_read', 'alerts_all'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const observabilityMinimalAll: Role = {
+  name: 'obs_minimal_read_spaces_all',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['minimal_all'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
+        },
+        spaces: ['space1'],
+      },
+    ],
+  },
+};
+
+export const observabilityMinimalAllSpacesAll: Role = {
+  name: 'obs_minimal_read_spaces_all',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['minimal_all'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const observabilityOnlyAlertsAll: Role = {
+  name: 'obs_alerts_read_spaces_all',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['alerts_all'],
```

Comment:
This isn't a realistic test scenario. A role can't grant the `alerts_read` sub-feature privilege without `minimal_read` or `minimal_all`.

Since you already have the `observabilityMinReadAlertsAll ` role defined above, I think this role and its associated test cases can be removed.